rust/src/builders/tx_builder.rs

```diff
@@ -426,17 +425,16 @@ impl TransactionBuilder {
         Ok(())
     }
 
-    fn cip2_random_improve_by<F>(
+    fn cip2_random_improve_by<F, R: Rng + ?Sized>(
         &mut self,
         available_inputs: &[TransactionUnspentOutput],
         available_indices: &mut BTreeSet<usize>,
         input_total: &mut Value,
         output_total: &mut Value,
         by: F,
-        rng: &mut rand::rngs::ThreadRng) -> Result<(), JsError>
+        rng: &mut R) -> Result<(), JsError>
```

Comment:
By changing the type to `rand::Rng` trait it is possible to use `ChaChaRng` from constant seeds in unit tests. However, current `ChaChaRng` of `rand_chacha` implements [`rand_os::rand_core::RngCore`](https://github.com/dcSpark/cardano-multiplatform-lib/blob/d9ec0ce32136deda2a497dd1c04e6dce8932ba5a/rust/src/chain_crypto/testing.rs#L6) instead of `rand::RngCore`, which cannot pass the [restriction](https://docs.rs/rand/latest/rand/trait.Rng.html) of `: rand::RngCore`. Any idea how to get around of this without upgrading? @SebastienGllmt

Reply:
yeah I remember all these crates have to be aligned at the same version to work properly by design so the upgrade is required